timeseries/tests/conftest.py

```diff
@@ -27,51 +34,46 @@ def pytest_collection_modifyitems(config, items):
             item.add_marker(skip_slow)
 
 
-@pytest.fixture()
-def temp_model_path(tmp_path_factory):
-    fn = tmp_path_factory.mktemp(str(uuid4())[:6])
-    return str(fn)
-
+def pytest_sessionstart():
+    """The following is a workaround to cache the dependencies from Hugging Face Hub once and
+    run the test session with HF_HUB_OFFLINE, i.e., preventing any HTTP calls from Hugging Face.
 
-@pytest.fixture(scope="session")
-def hf_model_path(tmp_path_factory):
-    """Force Hugging Face to cache the model config once and reuse it from a temporary cache directory.
-    This prevents inflating Hugging Face download numbers as an HTTP request is sent every time
-    ``ChronosPipeline.from_pretrained`` is called.
+    The code first calls `from_pretrained` in order to download and cache the two models (if they aren't
+    cached already) in a subprocess, and then sets the HF_HUB_OFFLINE environment variable to True, in
+    order to prevent any calls from the main process. The caching has to be done in a subprocess due to
+    the way Hugging Face Hub works: if HF_HUB_OFFLINE=1 is set in the main process before the call to
+    transformers, then the models cannot be downloaded and cached. If it is set after the transformers call,
```

Comment:
Nit: should this be "before importing `transformers`" in both cases?

Reply:
I think this reads correct.